server/data/data-ldap/src/test/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepositoryTest.java

```diff
@@ -69,188 +73,620 @@ static void afterAll() {
+        }
+
+        @Override
+        public AbstractUsersRepository testee() {
+            return usersRepository;
+        }
 
         @Test
         void knownUserShouldBeAbleToLogInWhenPasswordIsCorrect() throws Exception {
-            ReadOnlyUsersLDAPRepository ldapRepository = startUsersRepository(ldapRepositoryConfiguration(ldapContainer));
-            assertThat(ldapRepository.test(JAMES_USER, PASSWORD)).isTrue();
+            assertThat(usersRepository.test(JAMES_USER, PASSWORD)).isTrue();
         }
 
         @Test
         void knownUserShouldNotBeAbleToLogInWhenPasswordIsNotCorrect() throws Exception {
-            ReadOnlyUsersLDAPRepository ldapRepository = startUsersRepository(ldapRepositoryConfiguration(ldapContainer));
-            assertThat(ldapRepository.test(JAMES_USER, BAD_PASSWORD)).isFalse();
+            assertThat(usersRepository.test(JAMES_USER, BAD_PASSWORD)).isFalse();
         }
 
         @Test
         void unknownUserShouldNotBeAbleToLogIn() throws Exception {
-            ReadOnlyUsersLDAPRepository ldapRepository = startUsersRepository(ldapRepositoryConfiguration(ldapContainer));
-            assertThat(ldapRepository.test(UNKNOWN, BAD_PASSWORD)).isFalse();
+            assertThat(usersRepository.test(UNKNOWN, BAD_PASSWORD)).isFalse();
         }
 
         @Test
         void unknownUserShouldNotBeAbleToLogInWhenPasswordIsCorrect() throws Exception {
-            ReadOnlyUsersLDAPRepository ldapRepository = startUsersRepository(ldapRepositoryConfiguration(ldapContainer));
-            assertThat(ldapRepository.test(UNKNOWN, PASSWORD)).isFalse();
+            assertThat(usersRepository.test(UNKNOWN, PASSWORD)).isFalse();
         }
 
         @Test
-        void knownUserShouldBeAbleToLogInWhenPasswordIsCorrectWithVirtualHosting() throws Exception {
-            ReadOnlyUsersLDAPRepository ldapRepository = startUsersRepository(ldapRepositoryConfigurationWithVirtualHosting(ldapContainer));
-            assertThat(ldapRepository.test(JAMES_USER_MAIL, PASSWORD)).isTrue();
+        void containsWithGetUserShouldBeTrue() throws Exception {
+            assertThat(usersRepository.contains(usersRepository.getUser(JAMES_USER_MAIL.asMailAddress()))).isTrue();
         }
 
+        @Override
         @Test
-        void testShouldListUsers() throws Exception {
-            ReadOnlyUsersLDAPRepository ldapRepository = startUsersRepository(ldapRepositoryConfigurationWithVirtualHosting(ldapContainer));
-            assertThat(ImmutableList.copyOf(ldapRepository.list()))
-                .containsOnly(JAMES_USER_MAIL);
+        public void isAdministratorShouldReturnTrueWhenConfiguredAndUserIsAdmin(TestSystem testSystem) throws Exception {
+            assertThat(testee().isAdministrator(testSystem.getAdmin())).isTrue();
         }
 
+        @Disabled("JAMES-3088 isAdministrator is case sensitive")
```

Comment:
Why ? because of LDAP ?

Reply:
my mistake, it's fine. I enabled this test